# Lab | Inferential statistics - ANOVA

In this activity, we will look at another example. Your task is to understand the problem and write down all the steps to set up ANOVA. After the next lesson, we will ask you to solve this problem using Python. Here are the steps that you would need to work on: - Null hypothesis - Alternate hypothesis - Level of significance - Test statistic - P-value - F table

**Context**
Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power. You can find the data anova_lab_data.xlsx file in the files_for_lab folder

- State the null hypothesis
- State the alternate hypothesis
- What is the significance level
- What are the degrees of freedom of model, error terms, and total DoF

## Import and read files

In [1]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
import scipy

In [2]:
df=pd.read_excel('anova_lab_data.xlsx')

In [3]:
df.head()

Power   Etching Rate
0  160 W          5.43
1  180 W          6.24
2  200 W          8.79
3  160 W          5.71
4  180 W          6.71

In [4]:
df.columns = ['power', 'etchingrate']

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   power        15 non-null     object 
 1   etchingrate  15 non-null     float64
dtypes: float64(1), object(1)
memory usage: 368.0+ bytes


In [6]:
df.groupby('power').sum()

etchingrate
power             
160 W        28.96
180 W        31.19
200 W        41.59

In [7]:
# describe all the data 
df.describe()

etchingrate
count    15.000000
mean      6.782667
std       1.228643
min       5.430000
25%       5.845000
50%       6.240000
75%       7.725000
max       9.200000

## Hypothesis 

**H0 Null hypothesis:** No results in the etching rate when power (in Watts) of the plasma beam is changed.

**Ha State the alternate hypothesis:** Different results in the etching rate when power(in Watts) of the plasma beam is changed.


## Anova

In [8]:
model = ols('etchingrate ~ C(power)', data = df).fit()

In [9]:
results_table = sm.stats.anova_lm(model)
results_table

df     sum_sq   mean_sq          F    PR(>F)
C(power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN

- go 2 across and 12 down
- 3.89 is the critical value
- 36.8789 is the f value
- 36.8789 > 3.89 
- f is greater: there is a significant difference

In [10]:
results_table.columns

Index(['df', 'sum_sq', 'mean_sq', 'F', 'PR(>F)'], dtype='object')

In [11]:
results_table['PR(>F)']

C(power)    0.000008
Residual         NaN
Name: PR(>F), dtype: float64

The P-value is 0.000008 : statistical significant (because is less than 0.05)

In [12]:
# calculate the critical value
# f takes dfn and dfd as shape parameters.
# (ppf) Percent point function 
critical_value = scipy.stats.f.ppf(q = 1 - 0.05, dfn = 2, dfd = 12)
print('The critical value is:', round(critical_value,4))

The critical value is: 3.8853


In [13]:
pwr_n = len(df['power'])
er_n = len(df['etchingrate'])
DOF = (len(df.index) - 1) * 2
# the same as DOF = ((pwr_n + er_n) - 2)
print('The degree of freedom is:', DOF)

The degree of freedom is: 28


## Conclusions

In [14]:
# Level of significance
# 0.05
# 95% confidence

- The P-value is 0.000008 (statistical significant because is less than 0.05))
- The F critical value is: 3.8853 
- The F is: 36.878955
- F > 3.8853 
- The F is higher than the critical value
- There is a significant difference when different powers